src/rust/ensogl/lib/core/src/gui/cursor.rs

```diff
@@ -480,6 +481,10 @@ impl Cursor {
                 Vector3(x,y,0.0)
             }));
 
+            scene_position <- screen_position.map(f!([scene](position) {
+                scene.screen_to_scene_coordinates(*position)
+            }));
```

Comment:
This is written in a way too complex way - you can skip curly braces, you can also skip the `[scene]` part, as your expression starts with `scene.`. In fact you may be able to even make it one liner like 

```
scene_position <- screen_position.map(f!((p) scene.screen_to_scene_coordinates(*p)));
```